subsys/bluetooth/audio/codec.c

```diff
@@ -79,8 +158,99 @@ uint8_t bt_audio_codec_cfg_get_val(const struct bt_audio_codec_cfg *codec_cfg, u
 	return param.data_len;
 }
 
+int bt_audio_codec_cfg_set_val(struct bt_audio_codec_cfg *codec_cfg, uint8_t type,
+			       const uint8_t *data, size_t data_len)
+{
+	CHECKIF(codec_cfg == NULL) {
+		LOG_DBG("codec_cfg is NULL");
+		return -EINVAL;
+	}
+
+	CHECKIF(data == NULL) {
+		LOG_DBG("data is NULL");
+		return -EINVAL;
+	}
+
+	CHECKIF(data_len == 0U) {
+		LOG_DBG("data_len is 0");
+		return -EINVAL;
+	}
+
+	for (uint16_t i = 0U; i < codec_cfg->data_len;) {
+		const uint8_t len = codec_cfg->data[i++];
+		const uint8_t data_type = codec_cfg->data[i++];
+		const uint8_t value_len = len - sizeof(data_type);
+
+		if (data_type == type) {
+			uint8_t *value = &codec_cfg->data[i];
+
+			if (data_len == value_len) {
+				memcpy(value, data, data_len);
+			} else {
+				const uint8_t *old_next_data_start = value + value_len + 1;
+				const uint8_t data_len_to_move =
+					codec_cfg->data_len -
+					(old_next_data_start - codec_cfg->data);
+				uint8_t *new_next_data_start = value + data_len + 1;
+				const int16_t diff = data_len - value_len;
```

Comment:
`size_t` difference versus storage in `int16_t` ?

Reply:
Not sure what you ask here - Can you elaborate?

Reply:
`data_len` is `size_t` in the substraction, is that ok?

Reply:
With the check added from https://github.com/zephyrproject-rtos/zephyr/pull/63040#discussion_r1339707192 it should be OK